compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -3015,8 +3016,14 @@ public void visit(BLangRecordLiteral recordLiteral) {
                 }
 
                 LinkedHashMap<String, BField> fieldsInRecordType = spreadExprRecordType.fields;
+                List<String> unescapedSpreadExprFieldNameList = new ArrayList<>();
```

Comment:
Do we need another list for this? Can't we update `fieldsInRecordType` to have the unescaped names instead?

Reply:
The thing is there is no easy way to update the keys of a `LinkedHashMap`. Also changing the original value may cause other issues in other stages.

Reply:
Rather than updating, what I meant was creating `fieldsInRecordType` here with the unescaped names as the keys instead of directly using `spreadExprRecordType.fields`.

Reply:
Updated